### ISSUE TITLE
resolve

### ISSUE THREAD

**Issue by 89z** [yescallop/fluent-uri-rs#5]
Is it possible to resolve one URI in the context of another? Given base:

~~~
https://volvoforums.com
~~~

and reference:

~~~
forum/
~~~

return:

~~~
https://volvoforums.com/forum/
~~~

Examples:

- https://docs.rs/url/2.3.1/url/struct.Url.html#method.join
- <https://docs.rs/iri-string/latest/iri_string/types/struct.RiReferenceStr.html#method.resolve_against>

**Comment by 89z** [comment]
ah, just found this:

~~~
- [ ] Reference resolution.
~~~

https://github.com/yescallop/fluent-uri-rs#roadmap

**Comment by yescallop** [comment]
Yes, this is exactly on the roadmap. TBH I think reference resolution should be put before URI building on that list, because it would be more straightforward to design the resolution API.

That said, the resolution algorithm can be a bit of a pain to implement correctly due to some potential defect in it. I'll give it a try if I'm in the right mood, and any help with this is very welcome.
